### DISCUSSION TITLE
Windows App Certification says Avalonia app is not DPI aware

### DISCUSSION THREAD

**Discussion by TangDave** [AvaloniaUI/Avalonia#17769] [Q&A]
I'm testing my Avalonia app with the Windows App Certification Kit (prior to submitting to the Microsoft Store) and I notice the warning about DPIAwarenessValidation.

I added this section to the app.manifest file of my application:

```
      <windowsSettings>
        <dpiAware xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">true</dpiAware>
        <dpiAwareness xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">PerMonitorV2</dpiAwareness>
      </windowsSettings>	
```

Even with this section I still get the DPI warning. How did others with apps in the Microsoft Store get through this issue?

**Comment by maxkatz6** [comment]
Avalonia does set DpiAwareness with recommended APIs, SetProcessDpiAwarenessContext:
https://github.com/AvaloniaUI/Avalonia/blob/be8edab05b8672d63b7b825c7b1e4cabe2d9188e/src/Windows/Avalonia.Win32/Win32Platform.cs#L256-L288

**Reply by TangDave** [reply]
Is this code something I should reproduce to get rid of the warning or is it already in Avalonia and a known issue with Windows App Certification?

**Reply by maxkatz6** [reply]
It's already in Avalonia.
You can write a ticket to the Windows Store support. They might have some idea what's wrong with app certification.

**Reply by StefanKoell** [reply]
Maybe MS is only looking for a specific entry in the manifest file? If so, you could try to provide a manifest with dpi awareness set and see if it passes:
https://learn.microsoft.com/en-us/windows/win32/hidpi/setting-the-default-dpi-awareness-for-a-process

**Comment by TangDave** [comment]
Tried that - no difference. Anyway, it’s only a warning so I’ll ignore it -
just curious about what’s others did to work around it. Thanks.

**Comment by TangDave** [comment]
Thanks - I’ll ignore it for now and take it up with Microsoft if it causes
a problem.
